cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -360,6 +361,15 @@ class SnappyUnifiedMemoryManager private[memory](
           return
         }
 
+        // Stop execution pool to grow beyond eviction percentage of heap.
+        if (memoryMode eq MemoryMode.ON_HEAP) {
+          if (executionPool.memoryUsed + extraMemoryNeeded > maxHeapExecutionSize) {
+            logWarning(s"MemoryManager can't allocate $numBytes bytes as it " +
+                s"would exceed maxHeapExecutionSize = $maxHeapExecutionSize")
+            return
+          }
+        }
+
```

Comment:
There is no concept of execution block eviction in line with storage eviction. Individual MemoryConsumer might implement spill() method. If they don't get enough memory then they might spill their data to disk. We should discuss which all SnappyData execution structures we can spill to disk and how.